mne/simulation/raw.py

```diff
@@ -0,0 +1,619 @@
+
+    # Get chpi freqs and reorder
+    if chpi:
+        hpi_freqs = np.array([x['custom_ref'][0]
+                             for x in info['hpi_meas'][0]['hpi_coils']])
+        n_freqs = len(hpi_freqs)
+        order = [x['number'] - 1 for x in info['hpi_meas'][0]['hpi_coils']]
+        assert np.array_equal(np.unique(order), np.arange(n_freqs))
+        hpi_freqs = hpi_freqs[order]
+        hpi_order = info['hpi_results'][0]['order'] - 1
+        assert np.array_equal(np.unique(hpi_order), np.arange(n_freqs))
+        hpi_freqs = hpi_freqs[hpi_order]
+
+    # Extract necessary info
+    picks = pick_types(info, meg=True, eeg=True)  # for simulation
+    meg_picks = pick_types(info, meg=True, eeg=False)  # for CHPI
+    fwd_info = pick_info(info, picks)
+    fwd_info['projs'] = []  # Ensure no 'projs' applied
+    logger.info('Setting up raw data simulation using %s head position%s'
+                % (len(dev_head_ts), 's' if len(dev_head_ts) != 1 else ''))
+
+    if isinstance(stc, VolSourceEstimate):
+        verts = [stc.vertices]
+    else:
+        verts = stc.vertices
+    src = _restrict_source_space_to(src, verts)
+
+    blink_bem = blink_rr = ecg_rr = chpi_rrs = None
+
+    # Simulate Blink, ECG, and head-movement artifacts
+    # Oscillate blink artifact between resting (17 bpm) and reading (4.5 bpm)
+    # http://www.ncbi.nlm.nih.gov/pubmed/9399231
+
+    # Create blink_bem (used for eog and blink artifact) if necessary
+    if ecg or blink:
+
+        # Figure out our cHPI, ECG, and Blink dipoles
+        dig = info['dig']
+        assert all([d['coord_frame'] == FIFF.FIFFV_COORD_HEAD
+                    for d in dig if d['kind'] == FIFF.FIFFV_POINT_HPI])
+        chpi_rrs = [d['r'] for d in dig if d['kind'] == FIFF.FIFFV_POINT_HPI]
+        R, r0 = fit_sphere_to_headshape(info, verbose=False)[:2]
+        R /= 1000.
+        r0 /= 1000.
+
+        blink_rr = [d['r'] for d in info['dig']
+                    if d['ident'] == FIFF.FIFFV_POINT_NASION][0]
+        blink_rr = blink_rr - r0
+        blink_rr = (blink_rr / np.sqrt(np.sum(blink_rr * blink_rr)) *
+                    0.98 * R)[np.newaxis, :]
+        blink_rr += r0
+        blink_bem = make_sphere_model(
+            r0, head_radius=R, relative_radii=(0.99, 1.), sigmas=(0.33, 0.33),
+            verbose=False)
+
+    if blink:
+        blink_rate = np.cos(2 * np.pi * 1. / 60. * times)
+        blink_rate *= 12.5 / 60.
+        blink_rate += 4.5 / 60.
+        blink_times = rng.rand(len(times)) < blink_rate / info['sfreq']
+        # vary amplitudes
+        blink_times = blink_times * (rng.rand(len(times)) + 0.5)
+        # Convolve blink times with kernel to get blink traces
+        blink_kernel = np.hanning(int(0.25 * info['sfreq']))
+        blink_data = np.convolve(blink_times, blink_kernel,
+                                 'same')[np.newaxis, :]
+        blink_data += rng.randn(blink_data.shape[1]) * 0.05
```

Comment:
@Eric89GXL, @agramfort, artifacts currently have random noise added to them. I was thinking of taking this out since random noise gets added when simulating sensor data from stcs. Does that sounds reasonable?